python/ray/train/tests/test_trainer.py

```diff
@@ -1,1301 +0,0 @@
-            ),
-        )
-
-    with pytest.raises(ValueError):
-        trainer.run(
-            train_func,
-            checkpoint_strategy=CheckpointConfig(
-                checkpoint_score_attribute="missing_attribute"
-            ),
-        )
-
-
-def test_world_rank(ray_start_2_cpus):
-    config = TestConfig()
-
-    def train_func():
-        return train.world_rank()
-
-    trainer = Trainer(config, num_workers=2)
-    trainer.start()
-    results = trainer.run(train_func)
-
-    assert set(results) == {0, 1}
-
-
-def test_torch_auto_unwrap(ray_start_2_cpus):
-    """Tests if underlying model from DDP is extracted when saving ckpt."""
-
-    def train_fn():
-        model = torch.nn.Linear(1, 1)
-
-        # Wrap in DDP.
-        model = train.torch.prepare_model(model)
-
-        # Save DDP wrapped model.
-        train.save_checkpoint(model=model)
-
-        # Report DDP wrapped model.
-        train.report(model=model)
-
-    num_workers = 2
-    trainer = Trainer("torch", num_workers)
-    trainer.start()
-
-    class ValidateEncodedCallback(TrainingCallback):
-        def handle_result(self, results, **info):
-            for result in results:
-                model = result["model"]
-                assert isinstance(model, torch.nn.Module) and not isinstance(
-                    model, torch.nn.parallel.DistributedDataParallel
-                )
-
-    trainer.run(train_fn, callbacks=[ValidateEncodedCallback()])
-
-    last_checkpoint = trainer.latest_checkpoint
-    model = last_checkpoint["model"]
-    assert isinstance(model, torch.nn.Module) and not isinstance(
-        model, torch.nn.parallel.DistributedDataParallel
-    )
-
-    trainer.shutdown()
-
-
-def test_torch_amp(ray_start_2_cpus):
-    def train_fn():
-        train.torch.accelerate(amp=True)
-        model = torch.nn.Linear(1, 1)
-        model = train.torch.prepare_model(model)
-
-        # Make sure model is serializable even with amp enabled.
-        return model.module
-
-    num_workers = 2
-    trainer = Trainer("torch", num_workers)
-    trainer.start()
-
-    trainer.run(train_fn)
-    trainer.shutdown()
-
-
-def test_torch_amp_with_custom_get_state(ray_start_2_cpus):
-    """Tests amp with a model that has a custom __getstate__ method defined.
-
-    See https://discuss.ray.io/t/ray-train-hangs-for-long-time/6333/7
-    """
-
-    def train_fn():
-        train.torch.accelerate(amp=True)
-
-        class CustomLinear(torch.nn.Linear):
-            def __getstate__(self):
-                return self.__dict__.copy()
-
-        model = CustomLinear(1, 1)
-        model = train.torch.prepare_model(model)
-
-        # Make sure model is serializable even with amp enabled.
-        return model.module
-
-    num_workers = 2
-    trainer = Trainer("torch", num_workers)
-    trainer.start()
-
-    trainer.run(train_fn)
-    trainer.shutdown()
-
-
-def test_horovod_simple(ray_start_2_cpus):
-    import horovod.torch as hvd_torch
-
```

Comment:
this is already being tested in AIR right?

Reply:
Correct